src/messages/direct.rs

```diff
@@ -18,14 +18,15 @@ use crate::{
     ConnectionInfo,
 };
 use bincode::serialize;
+use serde::Serialize;
 use std::{
     fmt::{self, Debug, Formatter},
     hash::{Hash, Hasher},
-    mem,
+    ops::Deref,
 };
 
 /// Body of DirectVariant
-#[derive(Eq, PartialEq, Serialize, Deserialize)]
+#[derive(Eq, PartialEq, Hash, Serialize, Deserialize)]
```

Comment:
not 100% sure that was a win,  remove fairly simple implementation, but traded off a less easy usage.

Reply:
The custom `Hash` impl becomes more annoying when we merge `DirectVariant` and `RoutingVariant` into one type. Anyway, ideally we would have `Hash` implemented for `parsec::{Request, Response}`, then none of this would be necessary.

Reply:
Make sense, it only require to add Hash to parsec NetworkEvent since we already implement it anyway in Routing.